src/pages/Signup.js

```diff
@@ -11,17 +11,44 @@ import Typography from "@mui/material/Typography"
 import Container from "@mui/material/Container"
 import { createTheme, ThemeProvider } from "@mui/material/styles"
 import { Link as RouterLink } from "react-router-dom"
+import firebase from "../Firebase/FirebaseConfig"
 
 const theme = createTheme()
 
 export default function SignUp() {
+  const [signup, setSignup] = React.useState({
+    firstName: "",
+    email: "",
+    password: "",
+  })
+  console.log("signup", signup)
+
+  const handleChange = (e) => {
+    setSignup({
+      ...signup,
+      [e.target.name]: e.target.value,
+    })
+  }
+
   const handleSubmit = (event) => {
     event.preventDefault()
-    const data = new FormData(event.currentTarget)
-    console.log({
-      email: data.get("email"),
-      password: data.get("password"),
-    })
+    //authenticates a user signing up
+    firebase
+      .auth()
+      .createUserWithEmailAndPassword(signup.email, signup.password)
+      .then((userCredential) => {
+        // Signed in
+        var user = userCredential.user
+        console.log("user", user)
+        //adds a user to the firebase users collection
+        firebase.firestore().collection("users").doc(user.uid).set({
+          user: user.email,
+        })
+      })
+      .catch((error) => {
+        var errorCode = error.code
+        var errorMessage = error.message
+      })
```

Comment:
If the user is successfully created it would be cool if they were automatically logged in and redirected out of the sign up page. This could be covered in a separate task if it isn't already.

Reply:
I think this is a good thought that can be applied to the login form as well.

Reply:
I just added the redirect upon successful signup to the latest commit on this branch